lib/phoenix_live_view/html_formatter.ex

```diff
@@ -0,0 +1,424 @@
+defmodule Phoenix.LiveView.HTMLFormatter do
+  @moduledoc """
+  Format Heex templates from `.heex` files or `~H` sigils.
+
+  This is a `mix format` [plugin](https://hexdocs.pm/mix/main/Mix.Tasks.Format.html#module-plugins).
+
+  ## Setup
+
+  Add it as plugin to your `.formatter.exs` file and make sure to put the`heex` extension in
+  the `inputs` option.
+
+  ```elixir
+  [
+    plugins: [Phoenix.LiveView.HTMLFormatter],
+    inputs: ["*.{heex,ex,exs}", "priv/*/seeds.exs", "{config,lib,test}/**/*.{heex,ex,exs}"],
+    # ...
+  ]
+  ```
+
+  ## Options
+
+  * `:line_length` - The Elixir formatter defaults to a maximum line length
+    of 98 characters, which can be overwritten with the `:line_length` option
+    in your `.formatter.exs` file.
+
+  * `:heex_line_length` - change the line length only for the HEEx formatter.
+
+    ```elixir
+    [
+      # ...omitted
+      heex_line_length: 300
+    ]
+    ```
+
+  ## Formatting
+
+  This formatter tries to be as consistent as possible with the Elixir formatter.
+
+  Given HTML like this:
+
+  ```eex
+    <section><h1>   <b>Hello</b></h1> </section>
+  ```
+
+  It will be formatted as:
+
+  ```eex
+  <section>
+    <h1><b><%= @user.name %></b></h1>
+  </section>
+  ```
+
+  A block element will go to the next line, while inline elements will be kept in the current line
+  as long as they fit within the configured line length.
+
+  The following links list all block and inline elements.
+
+  * https://developer.mozilla.org/en-US/docs/Web/HTML/Block-level_elements#elements
+  * https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements#list_of_inline_elements
+
+  It will also keep inline elements in their own lines if you intentionally write them this way:
+
+  ```eex
+  <section>
+    <h1>
+      <b><%= @user.name %></b>
+    </h1>
+  </section>
+  ```
+
+  This formatter will place all attributes on their own lines when they do not all fit in the
+  current line. Therefore this:
+
+  ```eex
+  <section id="user-section-id" class="sm:focus:block flex w-full p-3" phx-click="send-event">
+    <p>Hi</p>
+  </section>
+  ```
+
+  Will be formatted to:
+
+  ```eex
+  <section
+    id="user-section-id"
+    class="sm:focus:block flex w-full p-3"
+    phx-click="send-event"
+  >
+    <p>Hi</p>
+  </section>
+  ```
+
+  ### Intentional new lines
+
+  The formatter will keep intentional new lines. However, the formatter will
+  always keep a maximum of one line break in case you have multiple ones:
+
+  ```eex
+  <p>
+    text
+
+
+    text
+  </p>
+  ```
+
+  Will be formatted to:
+
+  ```
+  <p>
+    text
+
+    text
+  </p>
+  ```
+  """
+
+  alias Phoenix.LiveView.HTMLAlgebra
+  alias Phoenix.LiveView.HTMLTokenizer
+
+  # Default line length to be used in case nothing is specified in the `.formatter.exs` options.
+  @default_line_length 98
+
+  @behaviour Mix.Tasks.Format
```

Comment:
We should check if we are using Elixir v1.13 or later and only add the @behaviour call in said versions. Something like:

    if Version.match? System.version(), ">= 1.13.0" do